holidays/ical.py

```diff
@@ -0,0 +1,140 @@
+        else:
+            self.ical_timestamp = ical_timestamp
+
+        try:
+            with open("holidays/version.py") as file:
+                for line in file:
+                    if line.startswith("__version__"):
+                        self.holidays_version = line.split("=")[1].strip().strip('"')
+                        break
+                else:
+                    raise ValueError("Version not found in holidays/version.py")
+        except FileNotFoundError:
+            raise FileNotFoundError("The file 'holidays/version.py' was not found.")
+
+        self.language = self._validate_language(getattr(self.holidays, "language", "EN"))
+
+    def _validate_language(self, language):
+        """
+        Validates the language code. Ensures it's a two-letter ISO 639-1 code
+        as iCal only supports that at the moment.
+
+        Args:
+        - language: The language code to validate.
+
+        Returns:
+        - Valid two-letter language code in uppercase.
+        """
+        language = (language or "EN").upper().split("_")[0]
+
+        if len(language) != 2:
+            raise ValueError(
+                f"Invalid language code: {language}. "
+                "Only two-letter ISO 639-1 codes supported by iCal"
+            )
+        return language
+
+    def _fold_line(self, line):
+        """
+        Fold long lines according to RFC 5545.
+
+        Lines of text SHOULD NOT be longer than 75 octets.
+        """
+        if len(line.encode("utf-8")) <= 75:
+            return [line]
+
+        result = [line]
+        while len(result[-1].encode("utf-8")) > 75:
+            current = result.pop()
+            pos = 74
+            while len(current[:pos].encode("utf-8")) > 74:
+                pos -= 1
+            result.append(current[:pos])
+            result.append(f" {current[pos:]}")
+
+        return result
+
+    def _generate_event(self, date, holiday_name):
+        """
+        Generate a single holiday event.
+
+        Args:
+        - date: Holiday date.
+        - holiday_name: Holiday name.
+
+        Returns:
+        - list[str]: List of iCalender format event lines.
+        """
+        formatted_date = date.strftime("%Y%m%d")
+        event_uid = f"{uuid.uuid4()}@{self.holidays_version}.holidays.local"
+
+        lines = [
+            "BEGIN:VEVENT",
+            f"DTSTAMP:{self.ical_timestamp}",
+            f"UID:{event_uid}",
+            f"SUMMARY:{holiday_name}",
+            f"DTSTART;VALUE=DATE:{formatted_date}",
+            "DURATION:P1D",
+            "END:VEVENT",
+        ]
+
+        folded_lines = []
+        for line in lines:
+            folded_lines.extend(self._fold_line(line))
+
+        return folded_lines
+
+    def generate(self):
+        """
+        Generate iCalendar data.
+
+        Yields:
+        - Union[str, bytes]: Each line of iCalendar format
+            (string or bytes depending on return_bytes).
```

Comment:
We need to return a single object (string or bytes) here, not a list. And line separator by RFC 5545 is CRLF.

Reply:
From what I understand, what's remaining is just `output = "\r\n".join(self.generate()))` and get the `.ics` file out 👀

The majority of the remaining work would be getting the same continuous holidays to use `DURATION` instead, and maybe adding `CATEGORIES` support.

Reply:
> From what I understand, what's remaining is just `output = "\r\n".join(self.generate()))` and get the `.ics` file out 👀

But it's better to deliver a finished product to the user. :)

> The majority of the remaining work would be getting the same continuous holidays to use `DURATION` instead, 

Hmm, that's an interesting idea. But it's not easy to implement.

> and maybe adding `CATEGORIES` support.

Yes, it would be very useful!

Reply:
> But it's better to deliver a finished product to the user. :)

Well, as I stated before I plan to include that in the new `.export` method instead of converting the existing `.generate` to that as the latter is still quite useful for testing 👀

Reply:
Implemented